.github/workflows/call.deploy-subgraph.yml

```diff
@@ -8,29 +8,28 @@ on:
         description: "Where to deploy subgraph to; superfluid, graph, or satsuma"
         type: string
       release_branch:
```

Comment:
need to start to be strict with the terminology to reduce confusion:

v1: it is a production environment; usually based of the "dev" branch; aiming for all networks
dev: it is a staging environment; usually based of a "tagged" release; aiming for all networks
feature: it is a testing environment, usually based of an active feature branch related to subgraph development; only one network

So, `deployment_environment` should sound fitting.

Reply:
Pay additional attention to the challenge that the "v1"/"dev" deployment environments of subgraph may have different github code branch.

Reply:
An example table:

| vendor:network \ subgraph deployment environment|  v1  | dev |
|:-----|:--------:|------:|
| the-graph:sepolia  | |  |
| satsuma:polygon   |  |  |

...

Reply:
Renaming from `release_branch` to `deployment_env` done in the modified files.
@0xdavinchee will do in other places as he's more familiar with that code.
The dev deployment in canary workflow (dev/gda) will also be handled outside of this PR.